projects/plugins/jetpack/_inc/lib/class.media-extractor.php

```diff
@@ -543,14 +549,21 @@ public static function get_images_from_html( $html, $images_already_extracted, $
 			}
 
 			if ( ! in_array( $queryless, $image_list, true ) ) {
-				if ( $extract_alt_text && ! empty( $extracted_image['alt_text'] ) ) {
-					$image_list[] = array(
-						'url'      => $queryless,
-						'alt_text' => $extracted_image['alt_text'],
-					);
+				$image_to_add = array(
+					'url' => $queryless,
+				);
+				if ( $extract_alt_text ) {
+					if ( ! empty( $extracted_image['alt_text'] ) ) {
+						$image_to_add['alt_text'] = $extracted_image['alt_text'];
+					}
+					if ( ! empty( $extracted_image['src_width'] ) || ! empty( $extracted_image['src_width'] ) ) {
+							$image_to_add['src_width']  = $extracted_image['src_width'];
+							$image_to_add['src_height'] = $extracted_image['src_height'];
+					}
 				} else {
-					$image_list[] = $queryless;
+					$image_to_add = $queryless;
```

Comment:
Shouldn't this also happen when we have `$extract_alt_text=true` but the `$extracted_image['alt_text']` is empty? Is this change intentional?

Reply:
Good question. I don't think it should. When `$extract_alt_text` is true, the calling function should be expecting an array of associative arrays. When it is false, it is expecting just an array of strings.